ndb_api/profile_helpers.go

```diff
@@ -86,15 +87,26 @@ func ResolveProfiles(ctx context.Context, ndb_client *ndb_client.NDBClient, data
 	// DB Param Profile
 	dbParam, err := dbParamProfileResolver.Resolve(ctx, dbEngineSpecific, DbParamOOBProfileResolver)
 	if err != nil {
-		log.Error(err, "DbParam Profile could not be resolved", "Input Profile", networkProfileResolver)
+		log.Error(err, "DbParam Profile could not be resolved", "Input Profile", dbParamProfileResolver)
 		return nil, err
 	}
 
+	// DB Param Instance Profile
+	dbParamInstance, err := dbParamInstanceProfileResolver.Resolve(ctx, dbEngineSpecific, DbParamOOBProfileResolver)
+	if err != nil {
+		// Database Parameter Instance profile is required only for sql server
+		if databaseType == common.DATABASE_TYPE_SQLSERVER {
+			log.Error(err, "Db Param Instance Profile could not be resolved", "Input Profile", dbParamInstanceProfileResolver)
+			return nil, err
+		}
```

Comment:
Also, I feel we should add an `else` block where we set the `err` to `nil`. 
Think of the scenario when we're provisioning a postgres db for example. We will not find a dbParamInstance profile. The resolver will return an error and we'll skip printing the error since it is not a MSSQL database. However, the err variable still holds an error and when the ResolveProfiles function returns, it will return the error in the `err` variable. 
WDYT?

Reply:
Makes sense! Done